src/plone/restapi/serializer/relationfield.py

```diff
@@ -33,4 +34,20 @@ class RelationChoiceFieldSerializer(DefaultFieldSerializer):
 @adapter(IRelationList, IDexterityContent, Interface)
 @implementer(IFieldSerializer)
 class RelationListFieldSerializer(DefaultFieldSerializer):
-    pass
+    def get_value(self, default=[]):
+        """Return field value reduced to list of non-broken Relationvalues.
+
+        Args:
+            default (list, optional): Default field value. Defaults to empty list.
+
+        Returns:
+            list: List of RelationValues
+        """
+        value = getattr(
+            self.field.interface(self.context), self.field.__name__, default
+        )
+        if not value:
+            return []
+        if isinstance(value, RelationValue):
+            return [value]
+        return [el for el in value if el.to_object]
```

Comment:
I think it's better to use `el.to_id` instead of `el.to_object`. `el.to_object` will wake up the objects.

Reply:
@wesleybl Ive tested this locally, turns otu el.to_id exists even if referenced to_id content has been deleted, so it ends up returning `[None]` as el.to_id couldnt filter it, so i guess awaking object is the only solution here.

Reply:
> Ive tested this locally, turns otu el.to_id exists even if referenced to_id content has been deleted

It exists but it returns None for the deleted content. So the item will not be added to the list.

Reply:
I saw that your test fails when `el.to_id` is used. This is because you missed calling `z3c.relationfield.event._setRelation`. For example:

```python
        rel1 = RelationValue(intids.getId(self.portal.doc1))
        rel2 = RelationValue(intids.getId(self.portal.doc2))
        self.portal.doc1.test_relationlist_field = [
            rel1,
            rel2,
        ]
        _setRelation(self.portal.doc1, "test_relationlist_field", rel1)
        _setRelation(self.portal.doc1, "test_relationlist_field", rel2)

```